src/index.js

```diff
@@ -1,10 +1,167 @@
+      document.getElementsByClassName('right-arr')[0].style.display = 'block'
+    }
+  }
+
+  /**
+   * This function will get the value from spinner  and update the card numbers
+   */
+  function getCardCount () {
+    cardCount = document.getElementById('select-icon').value
+    changeMiddleSegment(middleSegmentCards)
+    getScrollWidth()
+  }
+
+  /**
+   * This function will scroll the container when arrow is clicked
+   * @param {string} arrow It is used to identify the arrow
+   */
+  function scrollItems (arrow) {
+    const cardWidth = document.querySelector('.items-scroll').getBoundingClientRect().width
+    const scrollContainer = document.querySelector('.items-container')
+    if (arrow === 'right') scrollContainer.scrollLeft += cardWidth
+    else scrollContainer.scrollLeft -= cardWidth
+  }
+
+  /**
+   *  This function sorts the objects
+   * @param {object} data  all city details
+   *  @returns {object} returns the sorted city objects
+   */
+  function sortCityOptions (data) {
+    const sortedKeys = Object.keys(data).sort()
+    const result = { }
+    sortedKeys.forEach((key) => { result[key] = data[key] })
+    console.log(result)
+    return result
+  }
+
+  /**
+   *   This function selects the icon and add style. It calls the filter function
+   */
+  function selectIcon () {
+    if (selectedIcon !== ' ') { document.getElementById(selectedIcon).classList.remove('ic-click') }
+    selectedIcon = this.id
+    document.getElementById(this.id).classList.add('ic-click')
+    filterMiddleSegment(this.id)
+  }
+
+  /**
+   *  This function will filter the cities based on selected icons
+   * @param {string} iconId icon id for identification
+   */
+  function filterMiddleSegment (iconId) {
+    let filterObject = {}
+    document.getElementsByClassName('items-container')[0].innerHTML = ' '
+    Object.values(allCityDetails).map((data) => {
+      const temp = parseInt(data.temperature.slice(0, 2))
+      const humid = parseInt(data.humidity.slice(0, -1))
+      const precip = parseInt(data.precipitation.slice(0, -1))
+      const key = data.cityName.toLowerCase()
+      if (iconId === 'sunny-ic' && temp > 29 && humid < 50 && precip >= 50) {
+        filterObject[key] = allCityDetails[key]
+      } else if (iconId === 'snowflake-ic' && (temp >= 20 && temp <= 28) && humid > 50 && precip < 50) {
+        filterObject[key] = allCityDetails[key]
+      } else if (iconId === 'rainy-ic' && (temp < 20) && humid >= 50) {
+        filterObject[key] = allCityDetails[key]
+      }
+      return 0
+    }
+
+    )
+    if (iconId === 'sunny-ic')filterObject = compareFunction(filterObject, 'temperature')
+    else if (iconId === 'snowflake-ic')filterObject = compareFunction(filterObject, 'precipitation')
+    else filterObject = compareFunction(filterObject, 'humidity')
+    middleSegmentCards = filterObject
+    changeMiddleSegment(filterObject)
+    getScrollWidth()
+  }
+
+  const compareFunction = (obj, key) => {
```

Comment:
```suggestion
  const sortCitiesBasedOnWeatherType = (citiesObject, weatherType) => {
```
If it's to sort based on weather.